woocommerce/woocommerce-ios
# REST API: Migrate leaderboards endpoint

Part of #8544 


## Description

This PR migrates the leaderboards endpoint to enable REST API. Changes include:
- Updated `LeaderboardsListMapper` to parse contents without the data envelope
- Enabled REST API for the `loadLeaderboards` request.

## Testing instructions

- In `DefaultFeatureFlagService`, enable the flag for `applicationPasswordAuthenticationForSiteCredentialLogin` and build the app.
- Log out of the app or skip onboarding if needed.
- On the prologue screen, select "Enter your site address".
- Enter your test store address and tap "Continue".
- Enter the correct site credentials for your account in the test store.
- After the login succeeds, you should be navigated to the home screen.
- In Xcode console log, notice the event `🔵 Tracked dashboard_top_performers_loaded`. This means top performers have been loaded successfully.

Please feel free to test again by logging in with a WPCom account. The top performer fetch should still succeed. 

## Screenshots

N/A


---
